Sources/Logto/Core/LogtoCore.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  LogtoCore.swift
+//
+//
+//  Created by Gao Sun on 2022/1/18.
+//
+
+import Foundation
+
+enum LogtoCore {
+    private static let codeChallengeMethod = "S256"
+    private static let responseType = "authorization_code"
+    private static let prompt = "consent"
+
+    static func generateSignInUrl(
+        authorizationEndpoint: String,
+        clientId: String,
+        redirectUri: String,
+        codeChallenge: String,
+        state: String,
+        scope: ValueOrArray<String>? = nil,
+        resource: ValueOrArray<String>? = nil
+    ) throws -> URL {
+        guard
+            var components = URLComponents(string: authorizationEndpoint),
+            components.scheme != nil,
+            components.host != nil
+        else {
+            throw LogtoErrors.UrlConstruction.invalidAuthorizationEndpoint
+        }
+
+        let baseQueryItems = [
+            URLQueryItem(name: "client_id", value: clientId),
+            URLQueryItem(name: "redirect_uri", value: redirectUri.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)),
+            URLQueryItem(name: "code_challenge", value: codeChallenge),
+            URLQueryItem(name: "code_challenge_method", value: LogtoCore.codeChallengeMethod),
+            URLQueryItem(name: "state", value: state),
+            URLQueryItem(name: "scope", value:
+                (["openid", "offline_access"] + (scope?.inArray ?? [])).joined(separator: " ")),
```

Comment:
just want to align our usage of r default OIDC Provide scopes, following the SDK convention doc:
1. The is `scope` param defined in the `LogtoConfig` object for all platform SDKs. Which mentioned, `['openid', 'offline_access']` are always included. 
2. Here at the signinUrl generate methods or Core SDK also have a similar approach.  
should we keep two source of there scopes? cc: @xiaoyijun

Reply:
good call. i would prefer to remove mandatory `['openid', 'offline_access']` here and keep the logic in `LogtoConfig` init process.

Reply:
opened thread in slack

Reply:
I prefer to add the insurance logic code in both `core` and `LogtoCongifg`, for users may use them independently.